code/modules/admin/verbs/server.dm

```diff
@@ -22,7 +22,12 @@ ADMIN_VERB(toggle_hub, R_SERVER, "Toggle Hub", "Toggles the server's visilibilit
 #define HARDEST_RESTART "Hardest Restart (No actions, just reboot)"
 #define TGS_RESTART "Server Restart (Kill and restart DD)"
 ADMIN_VERB(restart, R_SERVER, "Reboot World", "Restarts the world immediately.", ADMIN_CATEGORY_SERVER)
-	var/list/options = list(REGULAR_RESTART, REGULAR_RESTART_DELAYED, HARD_RESTART, HARDEST_RESTART)
+	var/list/options = list(REGULAR_RESTART, REGULAR_RESTART_DELAYED, HARD_RESTART)
+
+	// this option runs a codepath that can leak db connections because it skips subsystem (specifically SSdbcore) shutdown
+	if(!SSdbcore.IsConnected())
+		options += HARDEST_RESTART
```

Comment:
I don't like limiting admin options, could we handle this in a warn instead of forbid manner or is that not feasible

Reply:
also, is this actually an us problem or is it just tts being bad at existence

Reply:
Nah, this is actually problematic and can cause db issues, b/c each instance of DD can eat up to 25 connections and using this option leaks them all.

Reply:
If we want to add it back we need to fix the underlying leak

Reply:
TGS restart (the 4th option) is just as effective anyway

Reply:
oook